README.md

```diff
@@ -1,833 +1,477 @@
+      - [Service Account Bearer Token Generation with Additional Context](#service-account-bearer-token-generation-with-additional-context)
+      - [Service Account Scoped Bearer Token Generation](#service-account-scoped-bearer-token-generation)
+      - [Skyflow Signed Data Tokens Generation](#skyflow-signed-data-tokens-generation)
     - [Logging](#logging)
   - [Reporting a Vulnerability](#reporting-a-vulnerability)
   
 ## Installation
 
-### Requirements
-- Node 12.22.12 and above
+#### Requirements
 
-### Configuration
+- Node 12.22.12 and above
 
 ```sh
 npm install skyflow-node
-
 ```
-## Usage
 
-### Importing `skyflow-node`
+#### Import / Require
 
-```
-const { Skyflow, generateBearerToken } = require('skyflow-node');
-```
-Or using ES modules
+**Require**
 
+```javascript
+const { Skyflow } = require('skyflow-node');
 ```
-import { Skyflow, generateBearerToken }  from 'skyflow-node';
+
+**ES modules**
+
+```javascript
+import { Skyflow }  from 'skyflow-node';
 ```
 
-### Service Account Bearer Token Generation
-The [service account](https://github.com/skyflowapi/skyflow-node/tree/master/src/service-account) module uses a credentials file to generate service account tokens. See [API Authentication](https://docs.skyflow.com/api-authentication/#create-a-service-account) for instructions on creating a service account.
+**All imports**
 
-The token generated from this module is valid for 60 minutes and lets you make API calls to the Data API as well as the Management API based on the permissions of the service account.
+```javascript
+import { Skyflow,                         // Vault client
+          generateBearerToken, isExpired, // JWT auth helpers
+          LogLevel, setLogLevel           // logging options
+} from 'skyflow-node'
+```
 
-The `generateBearerToken(filepath)` function takes the service account credentials file path for token generation. Alternatively, you can send the entire service account credentials as a string, by using `generateBearerTokenFromCreds(credentials)` function.
 
-Example using a service account credentials file path:
+## Vault APIs
+The [Vault](https://github.com/skyflowapi/skyflow-node/tree/master/src/vault-api) Node.js module is used to perform operations on the vault such as inserting records, detokenizing tokens, retrieving tokens for list of `skyflow_id`'s and to invoke the Connection.
+
+To use this module, the Skyflow client must first be initialized as follows: 
 
 ```javascript
-import {generateBearerToken, isExpired} from 'skyflow-node';
+import { Skyflow } from 'skyflow-node';
 
-const filepath = 'CREDENTIALS_FILE_PATH';
-let bearerToken = '';
+// Initialize the Skyflow client
+const vault = Skyflow.init({
+  // Id of the vault that the client should connect to.
+  vaultID: 'string',
+  // URL of the vault that the client should connect to.
+  vaultURL: 'string',
+  // Helper function generates a Skyflow bearer token.
+  getBearerToken: helperFunc,
```

Comment:
TODO: rename `helperFunc` to something like `authFunction` ?

Reply:
+1